README.md

```diff
@@ -18,115 +18,230 @@ and used in apps configured by Kopytko Packager.
+The `setState` method accepts also a callback function argument with the code to call once the component is updated
+and re-rendered.
+
+Please note that in an edge case, because children event listeners are set before the parent component is mounted,
+it is possible that the `setState` method is called before the component was mounted. In such case, state update
+will be delayed right after the `componentDidMount` lifecycle method call.
+
+An example usage of the component state in the render() method:
+```brightscript
+function render() as Object
+  return {
+    name: "Label",
+    props: {
+      id: "labelId",
+      text: m.state.text,
+    },
+  }
+end function
+```
 
 Looking at this example, you can see that the text prop of the `Label` element is set to the labelText property
 of the component state. The moment you call `setState({ labelText: "This text changed" })`, the label on the screen
 will be updated to reflect this state change, so you don't need to directly find the label element and change its text,
 everything is done automatically, the beauty of declarative programming. The same behaviour applies when using
-component prop values instead of the state. In this example, text could be bound to something like m.top.labelText,
+component prop values instead of the state. In this example, text could be bound to something like `m.top.labelText`,
 which is an interface field of the current component (a prop in Kopytko terminology), and the moment another parent
 component change the value of this field the same DOM update would happen again.
 
 ### Force update method
-The `forceUpdate()` method should be used when you need to use data from a source other than the component state or props.
-For example, you can have a global configuration that you can use in the render() method to assign some value to some
-element, but there's no way for Kopytko to know the source of this value and when it changes, so you can imperatively
-tell Kopytko to force an update when you know the value changed, which will check the current DOM tree with the new one
-(containing the updated global value now) and update the UI where needed. You can check the example in the following image
+The `forceUpdate` method should be used when you need to use data from a source other than the component state or props.
+For example, your Kopytko component is used as a list item component, so it doesn't have automatic observers on its props
+(because its parent is not a Kopytko component and therefore wasn't initialized via Kopytko mechanisms).
+You can imperatively tell Kopytko to force an update when you know the value changed, which will check the current
+DOM tree with the new one (containing the updated prop value now) and update the UI where needed. You can check the example below
 for a practical use case of this method.
 
-![img.png](docs/force-update.png)
+```brightscript
+sub init()
```

Comment:
Perhaps you should put a comment like: ` Assuming the component extends KopytkoGroup.xml`?